ddtrace/internal/datadog/profiling/stack_v2/src/stack_renderer.cpp

```diff
@@ -0,0 +1,89 @@
+#include "stack_renderer.hpp"
+#include "utf8_validate.hpp"
+
+using namespace Datadog;
+
+void
+StackRenderer::render_message(std::string_view msg)
+{
+    // This is unused for now, since the dd-trace-py profiler doesn't support emitting arbitrary messages.
+    (void)msg;
+}
+
+void
+StackRenderer::render_thread_begin(PyThreadState* tstate,
+                                   std::string_view name,
+                                   microsecond_t wall_time_us,
+                                   uintptr_t thread_id,
+                                   unsigned long native_id)
+{
+    (void)tstate;
+    sample = ddup_start_sample();
+    if (sample == nullptr) {
+        return;
+    }
+    ddup_push_threadinfo(sample, static_cast<int64_t>(thread_id), static_cast<int64_t>(native_id), name.data());
```

Comment:
stringview is not null terminated.  Shouldn't we also pass the length?

Reply:
Good call.  @r1viollet made the same observation.

Refactored interface.cpp to accept a `string_view` instead of a naked C string.